src/java/org/apache/cassandra/tools/NodeProbe.java

```diff
@@ -1734,10 +1734,11 @@ public long getReadRepairRepairedBackground()
     // JMX getters for the o.a.c.metrics API below.
     /**
      * Retrieve cache metrics based on the cache type (KeyCache, RowCache, or CounterCache)
-     * @param cacheType KeyCach, RowCache, or CounterCache
+     * @param type type, like Cache or UnweightedCache
+     * @param scope KeyCach, RowCache, or CounterCache
      * @param metricName Capacity, Entries, HitRate, Size, Requests or Hits.
      */
-    public Object getCacheMetric(String cacheType, String metricName)
+    public Object getCacheMetric(String type, String scope, String metricName)
```

Comment:
Since the `getCacheMetric` method isn't currently used for unweighted caches, would it be better to roll back these changes and do it when the nodetool needs it? 

Another option is to create an overloaded `public Object getCacheMetric` method that takes "Cache" as the first argument, this will help us avoid changes that are not used. The weighted and unweighted caches are slightly different anyway, so `getCacheMetric` is not quite what we want (e.g. the `size` is different for weighted and unweighted).

Reply:
I think this is leftover from times when I was trying to expose unweighted caches as well into nodetool's `Info` so it was not always called with "Cache" but I removed that functionality from `Info` because I just felt in my guts that this is not right thing to do anyway ... Metrics are going to be exposed via CQL and adding more metrics into nodetool just feels wrong. I can revert this whole change.